contracts/test/TestMintableCappedERC20.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+
+// solhint-disable no-unused-vars
+
+pragma solidity ^0.8.9;
+
+import { MintableCappedERC20 } from '../MintableCappedERC20.sol';
+
+contract InvalidMintableCappedERC20 is MintableCappedERC20 {
+    bool public shouldFailTransfer;
+
+    constructor(
+        string memory name,
+        string memory symbol,
+        uint8 decimals,
+        uint256 capacity
+    ) MintableCappedERC20(name, symbol, decimals, capacity) {
+        shouldFailTransfer = false;
+    }
+
+    function setFailTransfer(bool _shouldFailTransfer) public {
+        shouldFailTransfer = _shouldFailTransfer;
+    }
+
+    function transfer(address recipient, uint256 amount) public view override returns (bool) {
+        if (shouldFailTransfer) {
```

Comment:
is this extra contract needed? you can trigger a failure in the original contract using 0 address for recipient

Reply:
I couldn't think of a way around this contract. The recipient is hardcoded as `address(this)` referring to the gateway itself. Even if I could make the recipient the 0 address the transfer would revert and I would only be able to cover the first part of the `or` condition below:
` if (!success || (returnData.length != uint256(0) && !abi.decode(returnData, (bool)))) revert BurnFailed(symbol);`
Line 446

Reply:
Some ERC20 token implementations just return `false` when transfer fails instead of reverting. Maybe you would need to add such an ERC20 mock to cover that

Reply:
Yeah, I like that name better, just changed it